src/variants.php

```diff
@@ -699,18 +694,20 @@ function(sData) {
                 $_POST['id'] = $nID;
                 foreach($_POST['aTranscripts'] as $nTranscriptID => $aTranscript) {
                     if (!empty($_POST[$nTranscriptID . '_VariantOnTranscript/DNA']) && strlen($_POST[$nTranscriptID . '_VariantOnTranscript/DNA']) >= 6) {
-                        try {
-                            $oOutput = $_Mutalyzer->mappingInfo(array('LOVD_ver' => $_SETT['system']['version'], 'build' => $_CONF['refseq_build'], 'accNo' => $aTranscript[0], 'variant' => $_POST[$nTranscriptID . '_VariantOnTranscript/DNA']))->mappingInfoResult;
-                            if (isset($oOutput->errorcode)) {
-                                throw new Exception();
-                            }
-                            $_POST[$nTranscriptID . '_position_c_start'] = $oOutput->startmain;
-                            $_POST[$nTranscriptID . '_position_c_start_intron'] = $oOutput->startoffset;
-                            $_POST[$nTranscriptID . '_position_c_end'] = $oOutput->endmain;
-                            $_POST[$nTranscriptID . '_position_c_end_intron'] = $oOutput->endoffset;
-                        } catch (SoapFault $e) {
-                            lovd_soapError($e);
-                        } catch (Exception $e) {} // Pass when we get a "nice" Soap Error (variant not recognized, for instance).
+                        // 2017-09-22; 3.0-20; Replacing the old SOAP call to Mutalyzer with our new lovd_getVariantInfo() function.
+                        // Don't bother with a fallback, this thing is more solid than Mutalyzer's service.
+                        $aResponse = lovd_getVariantInfo($_POST[$nTranscriptID . '_VariantOnTranscript/DNA'], $aTranscript[0]);
+                        if ($aResponse) {
+                            $_POST[$nTranscriptID . '_position_c_start'] = $aResponse['position_start'];
+                            $_POST[$nTranscriptID . '_position_c_start_intron'] = $aResponse['position_start_intron'];
+                            $_POST[$nTranscriptID . '_position_c_end'] = $aResponse['position_end'];
+                            $_POST[$nTranscriptID . '_position_c_end_intron'] = $aResponse['position_end_intron'];
+                        } else {
+                            $_POST[$nTranscriptID . '_position_c_start'] = 0;
+                            $_POST[$nTranscriptID . '_position_c_start_intron'] = 0;
+                            $_POST[$nTranscriptID . '_position_c_end'] = 0;
+                            $_POST[$nTranscriptID . '_position_c_end_intron'] = 0;
```

Comment:
My gut feeling is that it's more clear to store unkown positions as NULL instead of 0.

Reply:
No, `NULL` is for variants where the position fields haven't been filled in.... so I need something else to indicate "we've looked at it, but I can't identify it". 0 seems the most logical choice...

Reply:
Shouldn't lines 680-681 then also be changed?

                 $_POST['position_g_start'] = NULL;
                 $_POST['position_g_end'] = NULL

Reply:
Yes, you're right, and also in 2449-2451. In some places this was used assuming the SOAP call had simply failed, and we should try again at some other time. Now we'll get a false only when the data isn't correct, so it should be changed. I'll take care of it, thanks for picking it out!

Reply:
Done in b0cdf0b.